server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -273,13 +295,34 @@ public void testExtensionsInitialize() throws Exception {
             );
 
             extensionsOrchestrator.extensionsInitialize();
-
-            transportService.close();
-            ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
             mockLogAppender.assertAllExpectationsMatched();
         }
     }
 
+    public void testHandleExtensionRequest() throws Exception {
+
+        Path extensionDir = createTempDir();
+
+        ExtensionsOrchestrator extensionsOrchestrator = new ExtensionsOrchestrator(settings, extensionDir);
+
+        extensionsOrchestrator.setTransportService(transportService);
+        extensionsOrchestrator.setClusterService(clusterService);
+        ExtensionRequest clusterStateRequest = new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_CLUSTER_STATE);
+        assertEquals(extensionsOrchestrator.handleExtensionRequest(clusterStateRequest).getClass(), ClusterStateResponse.class);
+
+        ExtensionRequest clusterSettingRequest = new ExtensionRequest(
+            ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_CLUSTER_SETTINGS
+        );
+        assertEquals(extensionsOrchestrator.handleExtensionRequest(clusterSettingRequest).getClass(), ClusterSettingsResponse.class);
+
+        ExtensionRequest localNodeRequest = new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_LOCAL_NODE);
+        assertEquals(extensionsOrchestrator.handleExtensionRequest(localNodeRequest).getClass(), LocalNodeResponse.class);
+
+        ExtensionRequest nullRequest = new ExtensionRequest(ExtensionsOrchestrator.RequestType.GET_SETTINGS);
+        assertEquals(extensionsOrchestrator.handleExtensionRequest(nullRequest), null);
```

Comment:
Lets not return `null`. Its going to make the system indeterministic when the caller doesnt handle it cleanly. 
Lets throw an exception and make sure this test expects that exception.